domain/domain.services.yml

```diff
@@ -9,6 +9,11 @@ services:
     arguments: ['@domain.negotiator']
     tags:
       - { name: 'context_provider' }
+  cache_context.domain:
+    class: Drupal\domain\Cache\DomainCacheContext
+    arguments: ['@request_stack']
```

Comment:
For reference, this would inject the @request_stack into the __consruct() method. That method isn't present, and we don't use the Request Stack, we  use the @domain_negotiator service.

You can see this pattern in https://github.com/agentrickard/domain/blob/8.x-1.x/domain_config/src/DomainConfigOverrider.php#L66